src/libAtomVM/opcodesswitch.h

```diff
@@ -3168,19 +3169,11 @@ static bool maybe_call_native(Context *ctx, AtomString module_name, AtomString f
                     TRACE("bs_append/7, fail=%i size=%li unit=%li src=0x%lx dreg=%c%i\n", fail, size_val, unit, src, T_DEST_REG(dreg_type, dreg));
 
                     size_t src_size = term_binary_size(src);
-                    uint8_t *buf = malloc(src_size);
-                    if (UNLIKELY(IS_NULL_PTR(buf))) {
-                        RAISE_ERROR(OUT_OF_MEMORY_ATOM);
-                    }
-                    memcpy(buf, term_binary_data(src), src_size);
-
                     if (UNLIKELY(memory_ensure_free(ctx, src_size + term_binary_data_size_in_terms(size_val / 8) + extra_val + BINARY_HEADER_SIZE) != MEMORY_GC_OK)) {
-                        free(buf);
                         RAISE_ERROR(OUT_OF_MEMORY_ATOM);
                     }
-                    term t = term_create_empty_binary(src_size + size_val / 8 + extra_val, ctx);
-                    memcpy((void *) term_binary_data(t), buf, src_size);
-                    free(buf);
+                    DECODE_COMPACT_TERM(src, code, i, src_off, src_off)
+                    term t = term_from_literal_binary(term_binary_data(src), src_size + size_val / 8 + extra_val, ctx);
```

Comment:
I don't think they are equivalent here.
master:
* term_create_empty_binary -> creates a binary filled with zeros which is `src_size + size_val / 8 + extra_val` bytes
* memcpy -> copies just `src_size` bytes into the newly created binary.
* `size_val / 8 + extra_val` are still empty.

your branch:
* term_from_literal_binary: creates a binary which is `src_size + size_val / 8 + extra_val` and copies into it `src_size + size_val / 8 + extra_val` byres regardless src size.

Reply:
Ah good point.  Corrected.